pytensor/tensor/elemwise.py

```diff
@@ -181,6 +181,16 @@ def __init__(self, input_broadcastable, new_order):
         self.augment = sorted([i for i, x in enumerate(new_order) if x == "x"])
         self.drop = drop
 
+        input_ndim = len(input_broadcastable)
+
+        self.is_left_expand_dims = "x" in new_order and (
+            input_ndim == 0 or new_order[:input_ndim] == list(range(input_ndim))
+        )
+
+        self.is_right_expand_dims = "x" in new_order and (
+            input_ndim == 0 or new_order[-input_ndim:] == list(range(input_ndim))
```

Comment:
This is simpler. The other was worrying about [:-0] slice 
```suggestion
            new_order[input_ndim:] == list(range(input_ndim))
```

Reply:
Same thing with augment as above

Reply:
Going to the right it needs to be different though yeah? In my example above `input_ndim = 2`  and `new_order = ('x', 0, 1)`, so `new_order[input_ndim:] = (1,)`, not `(0, 1)`

Reply:
Can I just check `new_order[0] == 'x' and new_order[1:] == list(range(input_ndim))` ?

Reply:
You could have multiple x on the left or right

Reply:
expand_dims left looks like `("x", "x", .. , "x", 0, 1, ..., n)`.

expand_dims right is the same but x are on the right

Reply:
My could should have been `order[:input_ndim]` not `order[input_ndim:]`

Reply:
updated

Reply:
If `inputs_ndim = 2` and `order = ('x', 0, 1)` (the case I'm looking at), neither of those work.

Reply:
Nothing was failing; you were worried about `-0` indexes. I am very lost.

Reply:
```python
is_left_expand_dims = augment and (inp_ndim == 0 or order[-inp_ndim:] == tuple(range(inp_ndim))
is_right_expand = augment and order[:inp_ndim] == tuple(range(inp_ndim))
```
Does that make sense?